src/sql/azdata.proposed.d.ts

```diff
@@ -670,10 +670,36 @@ declare module 'azdata' {
 		 */
 		delete?: boolean;
 	}
+
 	export interface DiffEditorComponent {
 		/**
 		 * Title of editor
 		 */
 		title: string;
 	}
+
+	export interface TextColumnOption {
+		content?: string
+	}
+
+	export interface RowDetailsOptions {
+		panelRows: number,
+		detailsHtml: boolean,
+		headerCssClass?: string,
+		dataPropertyName: string
+	}
+
+	export interface TableColumn {
+		field?: string
+	}
+
+	export interface TableComponentProperties extends ComponentProperties {
+		headerFilter?: boolean,
+		rowDetails?: RowDetailsOptions,
+		dataObjects?: any[]
```

Comment:
What's the difference supposed to be between data and dataObjects? It's going to be pretty confusing for users to have both I think - why couldn't you just set data and have the component handle that correctly

Reply:
I want to use "contentFormatter" that uses data object property passed through column option.

`{
	value: 'Message',
	field: 'message',
	headerCssClass: 'no-borders align-with-header',
	width: 900,
	options: {
		content: 'messageContent'
	}
}

private contentFormatter(dataValue: string, row: number | undefined, cell: any | undefined, value: any, columnDef: any | undefined, dataContext: any | undefined): string {
		let cellClasses = 'grid-cell-value-container';
		return `<div class="${cellClasses}">${dataContext[dataValue]}</div>`;
	}
`
This also allows me to specify property of the data object to display at rowDetails plugin.

Reply:
I don't see another way to display HTML here. It opens various opportunities like showing icon near the text.